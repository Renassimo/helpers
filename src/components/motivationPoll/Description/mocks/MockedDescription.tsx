const MockedDescription = jest.fn(() => <div>Mocked Description</div>);

export default MockedDescription;
