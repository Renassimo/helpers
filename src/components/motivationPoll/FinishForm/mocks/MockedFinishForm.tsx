const MockedFinishForm = jest.fn(() => <div>Mocked Finish Form</div>);

export default MockedFinishForm;
