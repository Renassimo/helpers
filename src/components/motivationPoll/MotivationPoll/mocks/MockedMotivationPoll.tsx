const MockedMotivationPoll = jest.fn(() => <div>MockedMotivationPoll</div>);

export default MockedMotivationPoll;
