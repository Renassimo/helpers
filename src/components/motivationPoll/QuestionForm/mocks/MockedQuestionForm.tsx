const MockedQuestionForm = jest.fn(() => <div>Mocked Question Form</div>);

export default MockedQuestionForm;
