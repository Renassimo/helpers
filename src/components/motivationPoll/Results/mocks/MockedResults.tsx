const MockedResults = jest.fn(
  ({ onDownloadPdf }: { onDownloadPdf: () => void }) => (
    <div>Mocked Results, onDownloadPdf: {!!onDownloadPdf}</div>
  )
);

export default MockedResults;
