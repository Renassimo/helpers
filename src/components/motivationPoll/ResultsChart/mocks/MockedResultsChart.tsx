const MockedResultsChart = jest.fn(() => <div>Mocked Results Chart</div>);

export default MockedResultsChart;
