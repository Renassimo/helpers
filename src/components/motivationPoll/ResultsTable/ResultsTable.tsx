import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';

const ResultsTable = () => {
  const { results } = useMotivationPoll();
  return (
    <Box my={2}>
      <TableContainer component={Paper}>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Number</TableCell>
              <TableCell>Description</TableCell>
              <TableCell>Median value</TableCell>
              <TableCell>Your result</TableCell>
            </TableRow>
          </TableHead>
          {results && (
            <TableBody>
              {results.map((demand) => (
                <TableRow key={demand.id}>
                  <TableCell>{demand.id + 1}</TableCell>
                  <TableCell>{demand.text}</TableCell>
                  <TableCell>{demand.medianValue}</TableCell>
                  <TableCell>{demand.points ?? 0}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          )}
        </Table>
      </TableContainer>
    </Box>
  );
};

export default ResultsTable;
