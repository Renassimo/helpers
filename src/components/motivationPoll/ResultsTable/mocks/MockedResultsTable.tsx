const MockedResultsTable = jest.fn(() => <div>Mocked Results Table</div>);

export default MockedResultsTable;
