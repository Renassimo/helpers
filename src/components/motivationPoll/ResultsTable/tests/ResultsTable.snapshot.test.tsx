import renderWithTheme from '@/tests/helpers/renderWithTheme';

import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import ResultsTable from '../ResultsTable';
import { mockedresults } from '@/types/motivationPoll/mocks';

jest.mock('@/hooks/motivationPoll/useMotivationPoll');

describe('ResultsTable snapshots', () => {
  beforeEach(() => {
    (useMotivationPoll as jest.Mock).mockImplementation(() => ({
      results: mockedresults,
    }));
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<ResultsTable />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when there are no results', () => {
    beforeEach(() => {
      (useMotivationPoll as jest.Mock).mockImplementation(() => ({
        results: null,
      }));
    });

    test('renders successfully', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(<ResultsTable />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });
});
