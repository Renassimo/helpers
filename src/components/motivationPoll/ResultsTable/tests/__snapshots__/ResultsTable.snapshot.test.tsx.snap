// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsTable snapshots renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1ebprri"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Number
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Median value
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Your result
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                First
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                3
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Second
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                4
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                6
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                4
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                Third
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                6
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                9
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;

exports[`ResultsTable snapshots when there are no results renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1ebprri"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Number
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Median value
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Your result
              </th>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </div>
</body>
`;
